python/rascaline/clib.py

```diff
@@ -15,8 +16,36 @@ def __call__(self):
             path = _lib_path()
             self._cache = cdll.LoadLibrary(path)
             setup_functions(self._cache)
+            _set_default_logging_callback()
         return self._cache
 
+def _set_default_logging_callback():
+    '''
+    Default logging function. For now print is used.
+    '''
+    def default_print(level, string):
+        print(string)
+    set_logging_callback(default_print)
+
+def set_logging_callback(function):
+    '''
+    Call `function` on every logging event. The callback should take a string
+    message and return nothing.
+
+    '''
+
+    def callback(level, message):
+        try:
+            function(str(level), message.decode("utf8"))
+        except Exception as e:
+            message = "exception raised in logging callback: {}".format(e)
+            warnings.warn(message, ResourceWarning)
+
+    global _CURRENT_CALLBACK
+    _CURRENT_CALLBACK = ctypes.CFUNCTYPE(None, ctypes.c_int, ctypes.c_char_p)(callback)
```

Comment:
Instead of rewriting the function type, could this import and use `RascalLoggingCallback` from _rascaline.py?

Reply:
hm... this works, I guess this is okay like this
```
    _CURRENT_CALLBACK = _get_library().rascal_set_logging_callback.argtypes[0](callback_wrapper)
```

Reply:
oh right. One simpler way to do so is to add 

```
rascal_logging_callback_t = ctypes.CFUNCTYPE(None, ctypes.c_int, ctypes.c_char_p)
```

around hand-written type definition https://github.com/Luthaf/rascaline/blob/cc7a5b98357825e45da8ac9b8e89453801f622f0/python/rascaline/_rascaline.py#L14-L19, 

and then use `rascal_logging_callback_t` in `rascal_set_logging_callback.argtypes`. This way, you can also `from _rascaline import rascal_logging_callback_t` where needed

Reply:
Not sure if this is simpler, because you need to change the the generation script in a transparent way. In the end it will be more code, might be more understandable but I am not convinced enough to go through it.